packages/@aws-cdk/aws-lambda/test/lambda-insights.test.ts

```diff
@@ -0,0 +1,195 @@
+import '@aws-cdk/assert-internal/jest';
+import { ResourcePart } from '@aws-cdk/assert-internal';
+import * as cdk from '@aws-cdk/core';
+import * as lambda from '../lib';
+
+// This test code has app.synth() because the lambda-insights code has functions that are only run on synthesis
+describe('lambda-insights', () => {
+  test('can provide arn to enable insights', () => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'Stack', {});
+    const layerArn = 'arn:aws:lambda:us-east-1:580247275435:layer:LambdaInsightsExtension:14';
+    new lambda.Function(stack, 'MyLambda', {
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_10_X,
+      insightsVersion: lambda.LambdaInsightsVersion.fromInsightVersionArn(layerArn),
+    });
+
+    expect(stack).toHaveResource('AWS::IAM::Role', {
+      ManagedPolicyArns:
+        [
+          { 'Fn::Join': ['', ['arn:', { Ref: 'AWS::Partition' }, ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole']] },
+          { 'Fn::Join': ['', ['arn:', { Ref: 'AWS::Partition' }, ':iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy']] },
+        ],
+    });
+
+    expect(stack).toHaveResource('AWS::Lambda::Function', {
+      Properties:
+      {
+        Code: { ZipFile: 'foo' },
+        Handler: 'index.handler',
+        Role: { 'Fn::GetAtt': ['MyLambdaServiceRole4539ECB6', 'Arn'] },
+        Runtime: 'nodejs10.x',
+        Layers: [layerArn],
+      },
+      DependsOn: ['MyLambdaServiceRole4539ECB6'],
+    }, ResourcePart.CompleteDefinition);
```

Comment:
I believe you're only verifying the `Layers` property, yes?

```suggestion
    expect(stack).toHaveResource('AWS::Lambda::Function', {
       Layers: [layerArn],
    });
```

Reply:
same in other tests.

Reply:
Done